lms/djangoapps/learner_recommendations/views.py

```diff
@@ -124,6 +128,71 @@ def get(self, request, course_id):
         )
 
 
+class CrossProductRecommendationsView(APIView):
+    """
+    **Example Request**
+
+    GET api/learner_recommendations/cross_product/{course_id}/
+    """
+
+    def _empty_response(self):
+        return Response({"courses": []}, status=200)
+
+    def _filter_empty_courses(self, courses):
+        for course in courses[:]:
+            if not course:
+                courses.remove(course)
+
+    def get(self, request, course_id):
+        """
+        Returns cross product recommendation courses
+        """
+        course_locator = CourseKey.from_string(course_id)
+        course_key = f'{course_locator.org}+{course_locator.course}'
+
+        associated_course_keys = get_cross_product_recommendations(course_key)
+
+        if associated_course_keys is None:
+            return self._empty_response()
+
+        fields = [
+            "key",
+            "uuid",
+            "title",
+            "owners",
+            "image",
+            "url_slug",
+            "course_type",
+            "course_runs",
+            "location_restriction",
+        ]
+        course_data = [get_course_data(key, fields) for key in associated_course_keys]
+        self._filter_empty_courses(course_data)
+
+        ip_address = get_client_ip(request)[0]
+        user_country_code = country_code_from_ip(ip_address).upper()
+
+        for course in course_data[:]:
```

Comment:
```suggestion
        for course in course_data:
```

Reply:
This is similar to the change where you asked to use list comprehension. Not looping over a copy when you're removing an item form the list you're looping from will result in an item in the iteration to be skipped. I was struggling to understand why the second course wasn't being removed in this list even though it also had a country restriction, but after reading up [here](https://www.python-engineer.com/posts/remove-elements-in-list-while-iterating/#what-not-to-do) I understood why.